uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/manager-leave-requests.html';
     }],
-    controllerAs: 'requests',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'Contact', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'OptionGroup', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, Contact, AbsencePeriod, AbsenceType, LeaveRequest, OptionGroup) {
     $log.debug('Component: manager-leave-requests');
+
+    var vm = Object.create(this),
+      allStatus = {
+        name: 'all',
+        label: 'All'
+      };
+
+    vm.isFilterExpanded = false;
+    vm.absencePeriods = [];
+    vm.filteredUsers = [];
+    vm.absenceTypes = [];
+    //leaveRequests.table - to handle table data
+    //leaveRequests.filter - to handle left nav filter data
+    vm.leaveRequests = {
+      table: {
+        list: []
+      },
+      filter: {
+        list: []
+      }
+    };
+    vm.filters = {
+      contactFilters: {
+        region: null,
+        department: null,
+        level_type: [],
+        location: null
+      },
+      leaveRequestFilters: {
+        selectedUsers: null,
+        selectedPeriod: null,
+        selectedAbsenceTypes: null,
+        leaveStatus: allStatus,
+        pending_requests: false
+      }
+    };
+    vm.pagination = {
+      page: 1,
+      size: 7
+    };
+    vm.loading = {
+      content: true,
+      page: true
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns the label of a Leave request when id is given
+     *
+     * @param {string} id - id of the leave request
+     * @return {string}
+     */
+    vm.getLeaveStatusByValue = function (id) {
+      if (vm.leaveRequestStatuses.length) {
```

Comment:
Is this `if` necessary, you think? Isn't this method supposed to be called when all the data had been fetched?